wpimath/src/main/java/edu/wpi/first/wpilibj/trajectory/Trajectory.java

```diff
@@ -206,6 +210,42 @@ public Trajectory relativeTo(Pose2d pose) {
             .collect(Collectors.toList()));
   }
 
+  /**
+   * Concatenates another trajectory to the current trajectory. The user is responsible for making
+   * sure that the end pose of this trajectory and the start pose of the other trajectory match (if
+   * that is the desired behavior).
+   *
+   * @param other The trajectory to concatenate.
+   * @return The concatenated trajectory.
+   */
+  @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+  public Trajectory concatenate(Trajectory other) {
+    // Deep copy the current states.
+    List<State> states =
+        m_states.stream()
+            .map(
+                state ->
+                    new State(
+                        state.timeSeconds,
+                        state.velocityMetersPerSecond,
+                        state.accelerationMetersPerSecondSq,
+                        state.poseMeters,
+                        state.curvatureRadPerMeter))
+            .collect(Collectors.toList());
+
+    for (int i = 1; i < other.getStates().size(); ++i) {
```

Comment:
We should add a similar comment as the C++ version to here regarding starting at i = 0 instead of i = 0.

Reply:
I added it, just to the wrong method :P